apps/jupyter/extensions/notebook-runner/src/iframe-api.ts

```diff
@@ -522,6 +580,129 @@ export class EmbeddedPythonCallbacks {
     }
     return files;
   }
+
+  private async writeCrtInternalTaskOnEditMode(
+    importedFiles: CrtInternalTask,
+  ): Promise<void> {
+    if (this.mode == Mode.edit) {
+      await this.putFileContents(
+        EmbeddedPythonCallbacks.taskTemplateLocation,
+        importedFiles.taskTemplate,
+      );
+
+      await this.createFolder(EmbeddedPythonCallbacks.dataLocation, "data");
+      await this.writeFolderContents(
+        EmbeddedPythonCallbacks.dataLocation,
+        importedFiles.data,
+      );
```

Comment:
Here and in other places: I think `writeFolderContents` is enough as it creates all folders.

Also, many of these calls are shared by both branches of the if/else. It will become much more readable if only the differences are in the if/else and the rest is before or after